Contrib/ChiralPairs/ChiralDescriptors.py

```diff
@@ -0,0 +1,300 @@
+    for d in range(mindist,maxdist+1):
+        if verbose:
+            print("Shell: ",d)
+        newShell = [n for n,i in enumerate(atomPaths) if i == d]
+        for aidx in newShell:
+            if verbose:
+                print("Atom ", aidx," in shell ",d)
+            atom = mol.GetAtomWithIdx(aidx)
+            # find neighbors of the current atom that are part of the substituent already
+            for n in atom.GetNeighbors():
+                nidx = n.GetIdx()
+                for k,v in sub_dict.items():
+                    # is the neighbor in the substituent and is not inthe same shell as the current atom 
+                    # and we haven't added the current atom already then put it in the correct substituent list
+                    if nidx in v and nidx not in newShell and aidx not in v:
+                        sub_dict[k].append(aidx)
+                        sharedNeighbors_dict[aidx]+=1
+                        maxShell_dict[k]=d
+                        if verbose:
+                            print("Atom ",aidx," assigned to ",nidx)
+    if verbose:
+        print(sub_dict)
+        print(sharedNeighbors_dict)
+        
+    return sub_dict, sharedNeighbors_dict, maxShell_dict
+
+def _getSizeOfSubstituents(sub, sharedNeighbors_dict, weighdownShared=True):
+    size=0
+    if weighdownShared:
+        for a in sub:
+            size+=1.0/sharedNeighbors_dict[a]
+        return size
+    return(len(sub))
+    
+def getBondsSubstituent(mol, atoms):
+    bonds=[]
+    for b in mol.GetBonds():
+        a1 = b.GetBeginAtomIdx()
+        a2 = b.GetEndAtomIdx()
+        if a1 in atoms and a2 in atoms:
+            bonds.append(b.GetIdx())
+    return bonds
+
+def getAromaticBondsSubstituent(mol, subAtoms):
+    bonds = getBondsSubstituent(mol, subAtoms)
+    aroBonds=0
+    for b in bonds:
+        if mol.GetBondWithIdx(b).GetIsAromatic():
+            aroBonds+=1
+    return aroBonds
+
+def getRotatableBondsSubstituent(mol, subAtoms):
+    rotatableBond = Chem.MolFromSmarts('[!$(*#*)&!D1]-&!@[!$(*#*)&!D1]')
+    matches = mol.GetSubstructMatches(rotatableBond)
+    numRotBonds=0
+    for a1,a2 in matches:
+        if a1 in subAtoms and a2 in subAtoms:
+            numRotBonds+=1
+    return numRotBonds
+
+
+substituentDescriptor = namedtuple('substituentDescriptor',
+                                   ['size','relSize','numNO','relNumNO','relNumNO_2','pathLength','relPathLength','relPathLength_2',
+                                    'sharedNeighbors', 'numRotBonds', 'numAroBonds'])    
+
+def calcSizeSubstituents(mol, sub_dict, sharedNeighbors_dict, maxShell_dict):  
+    sizeDict=defaultdict()
+    numAtoms = mol.GetNumAtoms()
+    #print(sharedNeighbors_dict)
+    for sidx, sub in sorted(sub_dict.items(), key= lambda x: len(x[1])):
+        size = _getSizeOfSubstituents(sub, sharedNeighbors_dict)
+        numNOs=0
+        numShared=0
+        # determine the number of oxygen and nitrogen atoms
```

Comment:
This comment seems not to reflect the code. `numNOs` is some weighted sum and not the count.